Add AWS WAF native

javiersanchz opened the pull request:
## Description

The native AWS WAF documentation has been added to master (`5.0.0`) for the issue https://github.com/wazuh/wazuh/issues/23364. 


## Checks
### Docs building
- [x] Compiles without warnings.
### Code formatting and web optimization
- [x] Uses three spaces indentation.
- [x] Adds or updates meta descriptions accordingly.
- [x] Updates the `redirects.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).
### Writing style
- [x] Uses present tense, active voice, and semi-formal registry.
- [x] Uses short, simple sentences.
- [x] Uses **bold** for user interface elements, _italics_ for key terms or emphasis, and `code` font for Bash commands, file names, REST paths, and code.

GGP1 commented on source/cloud-security/amazon/services/supported-services/waf.rst:
```suggestion
#. Click on *Create web ACL*:
```

GGP1 commented on source/cloud-security/amazon/services/supported-services/waf.rst:
```suggestion
#. Go to Services > Security, Identity, & Compliance > WAF & Shield:
```

GGP1 commented on source/cloud-security/amazon/services/supported-services/waf.rst:
```suggestion
#. Choose a name for your Web ACL and click on *Next*:
```

The next button is not shown in the image, perhaps we could do a little of zoom out and include it to be clear.

GGP1 commented on source/cloud-security/amazon/services/supported-services/waf.rst:
```suggestion
#. Click on *Add rules* and select which type of rule to add, then click on *Next*:
```

We are instructing the user to add a rule but we do not specify which or what to put in the form, is this expected?

javiersanchz replied:
Regarding the rules to implement, it was decided to keep it simple, showing the user where to add them because there are many ways to implement them.

Under "Add managed rule groups":

AWS managed rule groups
Cloudbric Corp. managed rule groups
Cyber Security Cloud Inc. managed rule groups
...
And under "Add my own rules":

Rule JSON editor
Rule visual editor
...
Therefore, adding a specific way of doing it didn't seem like the best option.

GGP1 commented on source/cloud-security/amazon/services/supported-services/waf.rst:
```suggestion
#. Set the rules priority and click on *Next*:
```

GGP1 commented on source/cloud-security/amazon/services/supported-services/waf.rst:
```suggestion
#. Choose a name for your CloudWatch metric and click on *Next*:
```

Where did we create this metric? If it is required we should instruct the user how to do it.

GGP1 commented on source/cloud-security/amazon/services/supported-services/waf.rst:
```suggestion
#. Review the Web ACL fields and click on *Create web ACL*  to create it:
```

GGP1 reviewed (changes requested):
We are repeating the `and click on *Next*` phrase on almost every step, perhaps it's not necessary since it's implied that the user should continue after having performed the action.

GGP1 reviewed (changes requested):
There are some requested changes from the previous review that weren't applied yet, please make sure to resolve them

GGP1 reviewed (dismissed):
LGTM

javiersanchz commented:
## Update

- Rebased with `master` to stay up to date
- Added a note in the WAF documentation about the new `waf_acls` parameter
- Ran without warnings
- The documentation is ready but will remain on hold until https://github.com/wazuh/wazuh/issues/23363 is resolved, in case we need to make any additional modifications

fdalmaup commented on source/cloud-security/amazon/services/supported-services/index.rst:
```diff
@@ -41,7 +41,7 @@ The next table contains the most relevant information about configuring each ser
 | Amazon       | :doc:`GuardDuty Firehose <guardduty>`                                         | bucket                | guardduty      | <bucket_name>/<pdocix>/<year>/<month>/<day>/<hh>                                                                 |
 +--------------+-------------------------------------------------------------------------------+-----------------------+----------------+------------------------------------------------------------------------------------------------------------------+
-| Amazon       | :doc:`WAF <waf>`                                                              | bucket                | waf            | <bucket_name>/<pdocix>/<year>/<month>/<day>/<hh>                                                                 |
+| Amazon       | :doc:`WAF <waf>`                                                              | bucket                | waf            | <bucket_name>/<prefix>/AWSLogs/<suffix>/<account_id>/WAFLogs/<region>/<waf_name>/<year>/<month>/<day>/<hh>/<mm>  |
```
```suggestion
| Amazon       | :doc:`WAF <waf>`                                                              | bucket                | waf            | <bucket_name>/<prefix>/AWSLogs/<suffix>/<account_id>/WAFLogs/<region>/<acl_name>/<year>/<month>/<day>/<hh>/<mm>  |
```

fdalmaup reviewed (changes requested):
The directory in which the images must be located is `/images/cloud-security/aws`.

fdalmaup commented on source/cloud-security/amazon/services/supported-services/waf.rst:
Given that the `Amazon configuration with Kinesis` will be deprecated, we should modify the `Wazuh configuration` section to contain the `waf_acls` parameter as part of the example.

fdalmaup reviewed (approved):
LGTM
